nms/app/packages/magmalte/app/views/equipment/GatewayPoolGatewaysEdit.js

```diff
@@ -49,22 +49,36 @@ export default function GatewayEdit(props: GatewayPoolEditProps) {
   const [error, setError] = useState('');
   const ctx = useContext(GatewayPoolsContext);
   const gwCtx = useContext(GatewayContext);
-  const [gwIds, _setGwIds] = useState(Object.keys(gwCtx.state));
   const [isPrimary, setIsPrimary] = useState(props.isPrimary || false);
   const [gwPool, setGwPool] = useState(props.gwPool);
   const [gateways, setGateways] = useState(
     isPrimary ? props.gatewayPrimary : props.gatewaySecondary,
   );
+  const gwIds = Object.keys(gwCtx.state);
 
+  // gateways already associated to gateway pools
+  const [gwPoolIds, setGwPoolIds] = useState(
+    Object.keys(ctx.state)
+      .map(poolId => {
+        return ctx.state[poolId].gatewayPoolRecords.map(
+          record => record.gateway_id,
+        );
+      })
+      .flat(),
+  );
```

Comment:
Thanks for adding the comment here, it's helpful. Please rename this variable though, it's not a `gateway pool id`, it's a `gateway id for gateway in a pool`
```suggestion
  // gateways already associated to gateway pools
  const [pooledGwIds, setPooledGwds] = useState(
    Object.keys(ctx.state)
      .map(poolId => {
        return ctx.state[poolId].gatewayPoolRecords.map(
          record => record.gateway_id,
        );
      })
      .flat(),
  );
```